packages/dapp-svelte-wallet/api/src/lib-wallet.js

```diff
@@ -633,15 +637,26 @@ export function makeWallet({
     return `instance ${q(petname)} successfully added to wallet`;
   };
 
-  const makeEmptyPurse = async (
+  /**
+   * This function is marked internal and unsafe because it permits importing a
+   * shared purse (which we don't necessarily trust).
+   *
+   * @param {Petname} brandPetname
+   * @param {Petname} petnameForPurse
+   * @param {boolean} defaultAutoDeposit
+   * @param {(issuer: Issuer) => ERef<Purse>} importPurse
+   */
+  const internalUnsafeImportPurse = async (
     brandPetname,
     petnameForPurse,
-    defaultAutoDeposit = false,
+    defaultAutoDeposit,
+    importPurse,
   ) => {
     const brand = brandMapping.petnameToVal.get(brandPetname);
     const { issuer } = brandTable.getByBrand(brand);
 
-    const purse = await E(issuer).makeEmptyPurse();
+    /** @type {Purse} */
+    const purse = await importPurse(issuer);
```

Comment:
This seems like a strange interaction compared to giving the purse itself.

Reply:
Good catch, I'll fix this.